### PR TITLE
Add test for MultiIndex.from_tuples

### LINKED ISSUE #23578: API/ERR/ENH: Allow MultiIndex.from_tuples to handle NaNs
This is the origin of #23558 and #23677, but IMO worthy to fix in its own right:

Trying to create a MultiIndex from a list of tuples containing NaNs (like `Index.str.partition` does if there are NaNs present) yields:
```
>>> pd.MultiIndex.from_tuples([('a', 'b', 'c'), np.nan, ('d', '', '')])
[...]
TypeError: object of type 'float' has no len()
```
However, it works easily when passing a *tuple* of NaNs
```
>>> pd.MultiIndex.from_tuples([('a', 'b', 'c'), (np.nan,) * 3, ('d', '', '')])
MultiIndex(levels=[['a', 'd'], ['', 'b'], ['', 'c']],
           labels=[[0, -1, 1], [1, -1, 0], [1, -1, 0]])
```
In fact, the length of the tuple is irrelevant:
```
>>> pd.MultiIndex.from_tuples([('a', 'b', 'c'), (np.nan,), ('d', '', '')])
MultiIndex(levels=[['a', 'd'], ['', 'b'], ['', 'c']],
           labels=[[0, -1, 1], [1, -1, 0], [1, -1, 0]])
```

Aside from the inconvenience, it is also a real problem to set elements of an array (say if there are several NaNs) to tuples, because they almost always get interpreted as another axis/list-like and then give creative errors.

I'm thinking this extra fail-safe should not be controversial, and I've got a fix prepared already, which is blocked by #23582 (at least in terms of adding tests).

### PR DESCRIPTION
**Pull request by GrammatikakisDimitris**
- [x] closes #23578
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/indexing/multiindex/test_multiindex.py [hunk 1]
```diff
@@ -206,3 +206,8 @@ def test_multiindex_with_na_missing_key(self):
         )
         with pytest.raises(KeyError, match="missing_key"):
             df[[("missing_key",)]]
+
+    def test_multiindex_from_tuples_with_nan(self):
+        # GH#23578
+        multi_idx = MultiIndex.from_tuples([("a", "b", "c"), np.nan, ("d", "", "")])
```
Comment (mroeschke) [inline pandas/tests/indexing/multiindex/test_multiindex.py:212]:
```suggestion
        result = MultiIndex.from_tuples([("a", "b", "c"), np.nan, ("d", "", "")])
```

File: pandas/tests/indexing/multiindex/test_multiindex.py [hunk 2]
```diff
@@ -206,3 +206,8 @@ def test_multiindex_with_na_missing_key(self):
...
+        assert isinstance(multi_idx, MultiIndex)
```
Comment (mroeschke) [inline pandas/tests/indexing/multiindex/test_multiindex.py:213]: Can you construct 

`expected = MultiIndex.from_tuples([("a", "b", "c"), (np.nan, np.nan, np.nan), ("d", "", "")])`

and then call

`tm.assert_index_equal_result, expected`

Comment (GrammatikakisDimitris) [reply pandas/tests/indexing/multiindex/test_multiindex.py]: Done

File: pandas/tests/indexing/multiindex/test_multiindex.py [hunk 3]
```diff
@@ -206,3 +206,11 @@ def test_multiindex_with_na_missing_key(self):
         )
         with pytest.raises(KeyError, match="missing_key"):
             df[[("missing_key",)]]
+
+    def test_multiindex_from_tuples_with_nan(self):
+        # GH#23578
+        MultiIndex.from_tuples([("a", "b", "c"), np.nan, ("d", "", "")])
```
Comment (mroeschke) [inline pandas/tests/indexing/multiindex/test_multiindex.py:212]:
```suggestion
        result = MultiIndex.from_tuples([("a", "b", "c"), np.nan, ("d", "", "")])
```

File: pandas/tests/indexing/multiindex/test_multiindex.py [hunk 4]
```diff
@@ -206,3 +206,11 @@ def test_multiindex_with_na_missing_key(self):
...
+        expected = MultiIndex.from_tuples(
+            [("a", "b", "c"), (np.nan, np.nan, np.nan), ("d", "", "")]
+        )
+        tm.assert_index_equal_result, expected
```
Comment (mroeschke) [inline pandas/tests/indexing/multiindex/test_multiindex.py:216]:
```suggestion
        tm.assert_index_equal(result, expected)
```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @GrammatikakisDimitris

**Pull request merged**